### PR TITLE
190 feat set up lambda for evaluate http

### PR DESCRIPTION
**Pull request by takeshi8989**
# Description
set up the process through | http -> evaluation publisher -> SQS -> evaluator

validate the request body in the evaluation publisher so we evaluate the candidates successfully or return error immediately

### CODE REVIEW

File: lambdas/checklist/serverless.yml [hunk 1]
```diff
@@ -71,5 +70,16 @@ functions:
               paths:
                 id: true
 
+  publish-evaluation:
```
Comment (kokiebisu) [inline lambdas/checklist/serverless.yml:73]: Maybe evaluation-publisher might be better. The other lambda that does processing can be named evaluation-processor.

File: lambdas/checklist/config.py [hunk 2]
```diff
@@ -11,6 +11,8 @@ class Config:
     POSTGRES_DB = os.environ.get('POSTGRES_DB')
     POSTGRES_USER = os.environ.get('POSTGRES_USER')
     POSTGRES_PASSWORD = os.environ.get('POSTGRES_PASSWORD')
+    
+    EVALUATION_QUEUE_URL = os.environ.get('EVALUATION_QUEUE_URL')
```
Comment (kokiebisu) [inline lambdas/checklist/config.py:15]: Can we name this EVALUATION_PROCESSOR_QUEUE_URL?

File: lambdas/checklist/evaluate.py [hunk 3]
```diff
@@ -235,20 +233,17 @@ def handler(event, context):
...
+        logger.info(f"Generated candidate evaluation successfully: {candidate_result}")
 
         save_candidate_evaluation_to_db(checklist_id, candidate_id, candidate_result)
-        logger.info("Saved candidate evaluation to database successfully")
         db_conn.commit()
-        return generate_success_response(candidate_result)
+        logger.info("Saved candidate evaluation to database successfully")
     except (ValueError, RuntimeError) as e:
         status_code = 400
         logger.error(f'Candidate evaluation failed (status {str(status_code)}): {e}')
```
Comment (kokiebisu) [inline lambdas/checklist/evaluate.py:243]: Do we still need the status code? Is this from the Github API?

File: lambdas/checklist/publish_evaluation.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
+import json
+import logging
+
+import boto3
+import psycopg2
+
```
Comment (kokiebisu) [inline lambdas/checklist/publish_evaluation.py:6]: Unnecessary blank line

File: lambdas/checklist/publish_evaluation.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+from config import Config
+
```
Comment (kokiebisu) [inline lambdas/checklist/publish_evaluation.py:9]: Unnecessary blank line

File: lambdas/checklist/serverless.yml [hunk 6]
```diff
@@ -30,6 +30,7 @@ provider:
     POSTGRES_DB: ${ssm:/terraform/${self:provider.stage}/rds/db_name}
     POSTGRES_USER: ${ssm:/terraform/${self:provider.stage}/rds/db_username}
     POSTGRES_PASSWORD: ${ssm:/terraform/${self:provider.stage}/rds/db_password}
+    EVALUATION_QUEUE_URL: ${ssm:/secrets/${self:provider.stage}/sqs/evaluation_queue_url}
```
Comment (kokiebisu) [inline lambdas/checklist/serverless.yml:33]: I will update the name to evaluation_processor_queue_url

File: lambdas/checklist/evaluate.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (takeshi8989) [reply lambdas/checklist/evaluate.py:243]: ?

File: lambdas/checklist/serverless.yml [hunk 7]
```diff
@@ -40,9 +41,7 @@ functions:
       command:
         - evaluate.handler
     events:
-      - http:
-          path: /checklists/evaluate
-          method: POST
+      - sqs: ${ssm:/terraform/${self:provider.stage}/sqs/criteria_evaluate_queue_arn}
```
Comment (kokiebisu) [inline lambdas/checklist/serverless.yml:44]: I will rename this to:
```
${ssm:/terraform/${self:provider.stage}/sqs/criteria_evaluation_processor_queue_arn}
```

File: lambdas/checklist/serverless.yml [hunk 8]
```diff
@@ -71,5 +70,16 @@ functions:
               paths:
                 id: true
 
+  publish-evaluation:
+    name: ${self:provider.stage}-checklist-publish-evaluation
```
Comment (kokiebisu) [inline lambdas/checklist/serverless.yml:74]: Let's update this to:
name: ${self:provider.stage}-checklist-evaluation-publisher

too

File: lambdas/checklist/serverless.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (takeshi8989) [reply lambdas/checklist/serverless.yml:33]: maybe checklist_evaluation_processor_queue_url is better?

File: lambdas/checklist/serverless.yml (hunk 7 above)
Comment (takeshi8989) [reply lambdas/checklist/serverless.yml:44]: maybe checklist_evaluation_processor_queue_ar is better?

**Pull request merged**